electrum/wallet.py

```diff
@@ -1845,6 +1845,25 @@ def get_new_sweep_address_for_channel(self) -> str:
         assert is_address(selected_addr), f"not valid bitcoin address: {selected_addr}"
         return selected_addr
 
+    def should_keep_reserve_utxo(self, to_be_spent_sat: int, is_anchor_channel_opening: bool) -> bool:
+        if not self.lnworker:
+            return False
+        is_reserve_needed = any(chan.has_anchors() and not chan.is_redeemed() for chan in self.lnworker.channels.values())
+        is_reserve_needed |= is_anchor_channel_opening
+        if not is_reserve_needed:
+            return False
+        available = sum(map(lambda c: c.value_sats(), self.get_spendable_coins()))
+        assert available - to_be_spent_sat >= 0
```

Comment:
is this assert sound?
e.g. consider this: wallet has 1 confirmed utxo (1 btc) and 1 unconfirmed utxo (1 btc). config has WALLET_SPEND_CONFIRMED_ONLY set to true. User uses manual coin selection to select both utxos, and clicks spend max when trying to open the channel. Then `to_be_spent_sat` I guess is 2 btc, and available is 1 btc.

so maybe call `wallet.get_spendable_coins()` with explicit `nonlocal_only=False`, `confirmed_only=False`

Reply:
right. I refactored  `should_keep_reserve_utxo` so that it takes `tx_inputs` and `tx_outputs`, and the available amount considers all coins in the wallet.